packages/workspace/src/tasks-runner/run-command.ts

```diff
@@ -21,19 +21,56 @@ import {
 } from '../utilities/project-graph-utils';
 import { output } from '../utilities/output';
 import { getDependencyConfigs } from './utils';
-import { Hasher } from '../core/hasher/hasher';
+import * as process from 'process';
+import { CompositeLifeCycle, LifeCycle } from './life-cycle';
+import { RunManyTerminalOutputLifeCycle } from './run-many-terminal-output-life-cycle';
+import { EmptyTerminalOutputLifeCycle } from './empty-terminal-output-life-cycle';
+import { RunOneTerminalOutputLifeCycle } from './run-one-terminal-output-life-cycle';
+import { TaskTimingsLifeCycle } from './task-timings-life-cycle';
 
 type RunArgs = yargs.Arguments & ReporterArgs;
 
+function getTerminalOutputLifeCycle(
+  initiatingProject: string,
+  terminalOutputStrategy: 'default' | 'hide-cached-output' | 'run-one',
+  projectsToRun: ProjectGraphNode[],
+  tasks: Task[],
+  nxArgs: NxArgs,
+  overrides: any
+) {
+  if (terminalOutputStrategy === 'run-one') {
+    return new RunOneTerminalOutputLifeCycle(
+      initiatingProject,
+      projectsToRun.map((t) => t.name),
+      tasks,
+      nxArgs
+    );
+  } else if (terminalOutputStrategy === 'hide-cached-output') {
+    return new EmptyTerminalOutputLifeCycle();
+  } else {
+    return new RunManyTerminalOutputLifeCycle(
+      projectsToRun.map((t) => t.name),
+      tasks,
+      nxArgs,
+      overrides
+    );
+  }
+}
+
 export async function runCommand<T extends RunArgs>(
   projectsToRun: ProjectGraphNode[],
   projectGraph: ProjectGraph,
   { nxJson, workspaceResults }: Environment,
   nxArgs: NxArgs,
   overrides: any,
-  reporter: Reporter,
+  terminalOutputStrategy: 'default' | 'hide-cached-output' | 'run-one',
```

Comment:
This union type is repeated twice inline in this file, might be nice to create a named type for it and reuse it